### ISSUE TITLE
Improve packaging

### ISSUE THREAD

**Issue by Kl4rry** [Kl4rry/simp#10]
Currently only raw binaries are distributed. This is not ideal especially for MacOS and Linux. It would be nice to package the app properly.

**Comment by Kl4rry** [comment]
Was a mistake in my derivation.

If I were to contribute `flake.nix` to the repo, would such a PR be accepted? I’m willing to support it.

_Originally posted by @GoldsteinE in https://github.com/Kl4rry/simp/issues/11#issuecomment-1231498023_

I happily accept help with package/platform/distro related stuff on other platforms then the ones I use as long as I don't have to maintain them as it is impossible for me to do so.

**Comment by GoldsteinE** [comment]
#13 Adds a flake for Nix (developer environment + package).

**Comment by Kl4rry** [comment]
Adding a [desktop entry](https://wiki.archlinux.org/title/desktop_entries) for linux would be nice.

**Comment by auronandace** [comment]
Perhaps cargo-binstall could be an option: https://github.com/cargo-bins/cargo-binstall/blob/main/SUPPORT.md

**Comment by Kl4rry** [comment]
Seems interesting. Can it run logic on installation like creating files like a [desktop entry](https://wiki.archlinux.org/title/desktop_entries) or is it just a `cargo install` for binaries?

**Comment by auronandace** [comment]
It is effectively cargo install for binaries, yes. Given the amount of dependencies for this project though does mean that you save some time and power installing it rather than having to compile it. I'd like more projects, especially the bigger ones, to support cargo-binstall.

**Comment by Kl4rry** [comment]
Yes I agree being able to install binaries would be nice considering both the long compile times and the problems with dependancies we have had.

**Comment by woelper** [comment]
I can recommend cargo-bundle to build a MacOS app bundle (I am building a [simple image viewer](https://github.com/woelper/oculante) with rudimentary mac support).

If you want to go further with file associations, it gets ugly fast, and I've been using `fruitbasket` with limited success. Here is my info.plist:
https://github.com/woelper/oculante/blob/master/res/info.plist

**Comment by Kl4rry** [comment]
I do not own a Mac computer and therefore is unable to debug bundling for Mac.
